src/main/java/picard/illumina/CollectIlluminaBasecallingMetrics.java

```diff
@@ -159,17 +174,19 @@ protected int doWork() {
                     barcodeToMetricCounts.put(barcode.toString(), new IlluminaMetricCounts(barcode.toString(), barcodeName, LANE));
                 }
             }
-
-            factory = barcodeToMetricCounts.isEmpty()
-                    ? new IlluminaDataProviderFactory(
+            if (USE_NEW_CONVERTER) {
+                factory = new IlluminaDataProviderFactory(BASECALLS_DIR, BARCODES_DIR, LANE, readStructure, bclQualityEvaluationStrategy);
```

Comment:
This assigns `factory` but its only use in the `USE_NEW_CONVERTER` case is inside another test for `USE_NEW_CONVERTER`. It seems like it would be simpler to only test `USE_NEW_CONVERTER` once and not use `factory for that case.